### PR TITLE
configure ci

### PR DESCRIPTION
**Pull request by tjjfvi**
I was setting up CI for hvm-core; figured it might be useful here too.

Configures GitHub Actions to automatically check/test/lint/fmt/spellcheck PRs.

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{
+  "version": "0.2",
+  "language": "en",
+  "words": [
+    "anni",
+    "annihilations",
+    "appllication's",
```
Comment (developedby) [inline cspell.json:7]: This is wrong appllication -> application.

File: cspell.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    "inodes",
+    "insta",
+    "interner",
+    "itertools",
+    "lcons",
+    "linearizes",
+    "linearizing",
+    "lnet",
+    "lnil",
+    "maximium",
```
Comment (developedby) [inline cspell.json:42]: maximium -> maximum

**Review by developedby — CHANGES_REQUESTED** [review]
I've never configured a github CI myself, so I'll trust you that the details are correct.

**Comment by developedby** [comment]
What is the spell-checking tool you're using here?

**Comment by tjjfvi** [comment]
Sorry, I reviewed the list of words for errors, but I guess I missed those two.

[cSpell](https://cspell.org/). I've used it a bit in the past, and have found it very helpful.

**Review by developedby — APPROVED** [review]

**Pull request merged**